File: modules/data/src/main/java/org/gjt/jclasslib/bytecode/AbstractBranchInstruction.java
package org.gjt.jclasslib.bytecode;

public abstract class AbstractBranchInstruction extends AbstractInstruction {

    private int branchOffset;

    /**
        Constructor.
        @param opcode the opcode.
     */
    protected AbstractBranchInstruction(Opcode opcode) {
        super(opcode);
    }

    /**
        Constructor.
        @param opcode the opcode.
        @param branchOffset the branch offset.
     */
    protected AbstractBranchInstruction(Opcode opcode, int branchOffset) {
        super(opcode);
        this.branchOffset = branchOffset;
    }

    /**
        Get the relative offset of the branch of this instruction.
        @return the offset
     */
    public int getBranchOffset() {
        return branchOffset;
    }

    /**
        Set the relative offset of the branch of this instruction.
        @param branchOffset the offset
     */
    public void setBranchOffset(int branchOffset) {
        this.branchOffset = branchOffset;
    }


}

File: modules/data/src/main/java/org/gjt/jclasslib/bytecode/AbstractBranchInstruction.kt
package org.gjt.jclasslib.bytecode

abstract class AbstractBranchInstruction(opcode: Opcode) : Instruction(opcode) {

    constructor(opcode: Opcode, branchOffset: Int) : this(opcode) {
        this.branchOffset = branchOffset
    }

    /**
     * Relative offset of the branch of this instruction.
     */
    var branchOffset: Int = 0

}
